pkg/traceability/muleemitter.go

```diff
@@ -2,89 +2,150 @@ package traceability
 
 import (
 	"encoding/json"
+	"fmt"
 	"time"
 
+	"github.com/Axway/agent-sdk/pkg/jobs"
+
+	"github.com/Axway/agent-sdk/pkg/util/log"
 	"github.com/sirupsen/logrus"
 
-	"github.com/elastic/beats/v7/libbeat/logp"
+	hc "github.com/Axway/agent-sdk/pkg/util/healthcheck"
 
 	"github.com/Axway/agents-mulesoft/pkg/anypoint"
 	"github.com/Axway/agents-mulesoft/pkg/config"
 )
 
+const healthCheckEndpoint = "ingestion"
+
 type Emitter interface {
-	Start()
-	Stop()
+	Start() error
 	OnConfigChange(gatewayCfg *config.AgentConfig)
 }
 
 // MuleEventEmitter - Represents the Gateway client
 type MuleEventEmitter struct {
-	client       anypoint.AnalyticsClient
-	cfg          *config.AgentConfig
-	done         chan bool
-	eventChannel chan string
-	pollInterval time.Duration
+	client            anypoint.AnalyticsClient
+	consecutiveErrors int
+	cfg               *config.AgentConfig
+	done              chan bool
+	eventChannel      chan string
+	pollInterval      time.Duration
+	jobID             string
+}
+
+// MuleEventEmitterJob wraps an Emitter and implements the Job interface so that it can be executed by the sdk.
+type MuleEventEmitterJob struct {
```

Comment:
Added a new struct that wraps the MuleEventEmitter. It implements the Emitter and the Job interface.